BioscoopSysteemAPI/BioscoopSysteemAPI/Controllers/PaymentController.cs

```diff
@@ -1,37 +1,183 @@
+            }
+        }
+
+        // GET: api/payments/1
+        /// <summary>
+        /// Get a payment by ID.
+        /// </summary>
+        /// <param name="id">Id of the payment.</param>
+        /// <returns>A payment object.</returns>
+        /// <response code="200">Succesfully returns a payment object.</response>
+        /// <response code="404">Error: The object you are looking for is not found.</response>
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [HttpGet("{id}")]
+        public async Task<ActionResult<PaymentReadDTO>> GetPayment(int id)
+        {
+            try
+            {
+                var domainPayment = await _paymentRepository.GetPaymentByIdAsync(id);
+                var dtoPayment = _mapper.Map<PaymentReadDTO>(domainPayment.Value);
+
+                if (dtoPayment == null)
+                {
+                    return NotFound();
+                }
+                return Ok(dtoPayment);
+            }
+            catch (Exception)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, "Error retrieving data from the database");
+            }
+        }
+
+        // DELETE: api/payment/1
+        /// <summary>
+        /// Delete an object by Id in the database.
+        /// </summary>
+        /// <param name="id">Id of the object.</param>
+        /// <returns></returns>
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [HttpDelete("{id}")]
+        public async Task<ActionResult<PaymentDeleteDTO>> DeletePayment(int id)
+        {
+            try
+            {
+                var payment = await _paymentRepository.DeletePaymentAsync(id);
+
+                if (payment == null)
+                {
+                    return NotFound();
+                }
+                return Ok(payment);
+            }
+            catch (Exception)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, "Error retrieving data from the database");
+            }
+        }
+
+        /// <summary>
+        /// Update a payment.
+        /// </summary>
+        /// <param name="id">Id of the payment</param>
+        /// <param name="payment">Payment object</param>
+        /// <returns>Action result without content.</returns>
+        /// <response code="204">Succesfully updated object.</response>
+        /// <response code="404">Error: The object you are looking for is not found.</response>
+        // To protect from overposting attacks, see https://go.microsoft.com/fwlink/?linkid=2123754
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [HttpPut("{id}")]
+        public async Task<ActionResult<Payment>> PutPayment(int id, Payment payment)
```

Comment:
Een betaling dat is gedaan wijzigen?? zouden we deze methode moeten hebben? wat denken jullie